Source/WebInspectorUI/UserInterface/Views/OverrideDeviceSettingsPopover.js

```diff
@@ -0,0 +1,324 @@
+                {name: `Safari ${emDash} iPadOS 16.0 ${emDash} iPad mini`, value: "Mozilla/5.0 (iPad; CPU iPhone OS 16_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.0 Mobile/15E148 Safari/604.1"},
+                {name: `Safari ${emDash} iPadOS 16.0 ${emDash} iPad`, value: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.0 Safari/605.1.15"},
+            ],
+            [
+                {name: `Microsoft Edge ${emDash} macOS`, value: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.53 Safari/537.36 Edg/103.0.1264.37"},
+                {name: `Microsoft Edge ${emDash} Windows`, value: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.53 Safari/537.36 Edg/103.0.1264.37"},
+            ],
+            [
+                {name: `Google Chrome ${emDash} macOS`, value: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.53 Safari/537.36"},
+                {name: `Google Chrome ${emDash} Windows`, value: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"},
+            ],
+            [
+                {name: `Firefox ${emDash} macOS`, value: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:101.0) Gecko/20100101 Firefox/101.0"},
+                {name: `Firefox ${emDash} Windows`, value: "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:101.0) Gecko/20100101 Firefox/101.0"},
+            ],
+            [
+                {name: WI.UIString("Other"), value: "other"},
+            ],
+        ];
+
+        let selectedOptionElement = null;
+
+        for (let group of userAgents) {
+            for (let {name, value} of group) {
+                let optionElement = userAgentValueSelect.appendChild(document.createElement("option"));
+                optionElement.value = value;
+                optionElement.textContent = name;
+
+                if (value === WI.deviceSettingsManager.overridenDeviceUserAgent)
+                    selectedOptionElement = optionElement;
+            }
+
+            if (group !== userAgents.lastValue)
+                userAgentValueSelect.appendChild(document.createElement("hr"));
+        }
+
+        let userAgentValueInput = null;
+
+        let showUserAgentInput = () => {
+            if (userAgentValueInput)
+                return;
+
+            userAgentValueInput = userAgentValue.appendChild(document.createElement("input"));
+            userAgentValueInput.spellcheck = false;
+            userAgentValueInput.value = userAgentValueInput.placeholder = WI.deviceSettingsManager.overridenDeviceUserAgent || navigator.userAgent;
```

Comment:
Aside: hmm i dunno how i never noticed this before, but this wont work for remote inspection.  we might want to do a
```
WI.runtimeManager.evaluateInInspectedWindow(`globalThis.navigator?.userAgent`, {returnByValue: true}, (result, wasThrown) => {
    if (!wasThrown && result)
        this._initialUserAgent = result;
});
```
inside `WI.DeviceSettingsManager.prototype.initializeTarget` 🤔

Reply:
I understand what the intent is. While it shouldn't technically matter what the free-form user agent field is populated with since the developer is going to replace it with whatever they want anyway, it would indeed be nice to offer the target's actual default user agent as a starting point.

I adapted your example. It needed an `objectGroup` and when using `returnByValue: true` option, the actual result is the third argument in the callback (the first argument is the represented object, which is `null` with this option [RuntimeManager.js#L144-L145](https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/UserInterface/Controllers/RuntimeManager.js#L144-L145))

```javascript
const options = {
    objectGroup: WI.DeviceSettingsManager.ObjectGroup,
    returnByValue: true,
}

WI.runtimeManager.evaluateInInspectedWindow("globalThis.navigator?.userAgent", options, (remoteObject, wasThrown, result) => {
    if (!wasThrown && result)
        this._deviceUserAgent = result.value;

    target.RuntimeAgent.releaseObjectGroup(WI.DeviceSettingsManager.ObjectGroup);
});
```

Not sure if `target.RuntimeAgent.releaseObjectGroup()` was necessary, but it seems like good hygiene since this group is used for just this one operation.